osu.Game/Screens/Ranking/Expanded/ExpandedPanelMiddleContent.cs

```diff
@@ -224,6 +224,8 @@ private void load(BeatmapDifficultyCache beatmapDifficultyCache)
             if (score.Date != default)
                 AddInternal(new PlayedOnText(score.Date));
 
+            var starDifficulty = beatmapDifficultyCache.GetDifficultyAsync(beatmap, score.Ruleset, score.Mods).GetResultSafely();
```

Comment:
This star difficulty struct already has max combo, does it not? So why not just use that? Is it because of the `score.BeatmapInfo.MaxCombo != null` fallback path in `ScoreManager`? If it is because of that, then maybe a better change would to be just remove that fallback path and fix `ScoresContainer`?

It just feels like going in circles with this bandaid fix on top of a todo that is going to have to be pretty much reverted anyway as is when `ScoresContainer` is fixed to work properly via interfaces.

Reply:
More like, according to the logic in `GetTotalScoreAsync`, `GetDifficultyAsync` returns "osu!stable combo values", which means lazer scores will receive incorrect maximum combo in that case.

Had mentioned that in OP as well:
> Since `StarDifficulty.MaxCombo` can't always be relied on due to being valid for stable/legacy scores only, I've exposed the local logic `GetTotalScoreAsync` has for computing max combo to its own `GetBeatmapMaximumComboAsync` method and used it instead.